### PR TITLE
Add more Control tests

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Add more tests for Control and some base classes

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2229)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/MonthCalendarTests.cs [hunk 1]
```diff
@@ -15,109 +15,113 @@ public class MonthCalendarTests
         [Fact]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/MonthCalendarTests.cs]:
```suggestion
        [WinFormsFact]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs [hunk 2]
```diff
@@ -5966,28 +6326,45 @@ public void Control_AllowDrop_Set_GetReturnsExpected(bool value)
...
             Assert.Equal(value, control.AllowDrop);
+            Assert.False(control.IsHandleCreated);
 
             // Set different.
             control.AllowDrop = value;
             Assert.Equal(value, control.AllowDrop);
+            Assert.False(control.IsHandleCreated);
         }
 
         [Fact]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs]:
```suggestion
        [WinFormsFact]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs [hunk 3]
```diff
@@ -5966,28 +6326,45 @@ public void Control_AllowDrop_Set_GetReturnsExpected(bool value)
...
         public void Control_AllowDrop_SetWithHandleNonSTAThread_ThrowsInvalidOperationException()
         {
             var control = new Control();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs]:
```suggestion
            using var control = new Control();
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Design/ComponentEditorPageTests.cs [hunk 4]
```diff
@@ -18,93 +18,99 @@ public class ComponentEditorPageTests
...
-            Assert.Empty(page.Title);
-            Assert.Equal(0, page.Top);
-            Assert.False(page.Visible);
-            Assert.NotNull(page.VerticalScroll);
-            Assert.Same(page.VerticalScroll, page.VerticalScroll);
-            Assert.False(page.VScroll);
-            Assert.Equal(200, page.Width);
-
-            Assert.False(page.IsHandleCreated);
+            var control = new SubComponentEditorPage();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Design/ComponentEditorPageTests.cs]:
```suggestion
            using var control = new SubComponentEditorPage();
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Design/ComponentEditorPageTests.cs [hunk 5]
```diff
@@ -18,93 +18,99 @@ public class ComponentEditorPageTests
         [Fact]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Design/ComponentEditorPageTests.cs]:
```suggestion
        [WinFormsFact]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs [hunk 6]
```diff
@@ -13597,22 +14168,272 @@ public void Control_RescaleConstantsForDpi_Invoke_Nop(int deviceDpiOld, int devi
...
+            // Invoke again.
             control.ResetMouseEventArgs();
+            Assert.False(control.IsHandleCreated);
         }
 
-        [Fact]
+        [WinFormsFact]
         public void Control_ResetMouseEventArgs_InvokeWithHandle_Success()
         {
             var control = new SubControl();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs]:
```suggestion
            using var control = new SubControl();
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs [hunk 7]
```diff
@@ -13597,22 +14168,272 @@ public void Control_RescaleConstantsForDpi_Invoke_Nop(int deviceDpiOld, int devi
...
+            
+            // Reset again.
+            control.ResetImeMode();
+            Assert.Equal(ImeMode.NoControl, control.ImeMode);
+        }
+
+        [WinFormsFact]
         public void Control_ResetMouseEventArgs_InvokeWithoutHandle_Success()
         {
             var control = new SubControl();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs]:
```suggestion
            using var control = new SubControl();
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs [hunk 8]
```diff
@@ -5996,17 +6373,35 @@ public void Control_AllowDrop_SetWithHandleSTA_GetReturnsExpected(bool value)
         {
             var control = new Control();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs]:
```suggestion
            using var control = new Control();
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs [hunk 9]
```diff
@@ -5966,28 +6326,45 @@ public void Control_AllowDrop_Set_GetReturnsExpected(bool value)
...
             // Can set to false.
             control.AllowDrop = false;
             Assert.False(control.AllowDrop);
+            Assert.True(control.IsHandleCreated);
+            Assert.Equal(0, invalidatedCallCount);
+            Assert.Equal(0, styleChangedCallCount);
+            Assert.Equal(0, createdCallCount);
         }
 
         [StaTheory]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs]:
```suggestion
        [WinFormsTheory]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ScrollBarTests.cs [hunk 10]
```diff
@@ -16,74 +16,79 @@ public class ScrollBarTests
...
-            Assert.Equal(Size.Empty, scrollBar.Size);
-            Assert.Equal(1, scrollBar.SmallChange);
-            Assert.Equal(0, scrollBar.TabIndex);
-            Assert.False(scrollBar.TabStop);
-            Assert.Empty(scrollBar.Text);
-            Assert.Equal(0, scrollBar.Top);
-            Assert.Equal(0, scrollBar.Value);
-            Assert.True(scrollBar.Visible);
-            Assert.Equal(0, scrollBar.Width);
+            var control = new SubScrollBar();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ScrollBarTests.cs]:
```suggestion
            using var control = new SubScrollBar();
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ScrollBarTests.cs [hunk 11]
```diff
@@ -16,74 +16,79 @@ public class ScrollBarTests
         [Fact]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ScrollBarTests.cs]:
```suggestion
        [WinFormsFact]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TabControlTests.cs [hunk 12]
```diff
@@ -1559,6 +1565,44 @@ public void TabControl_RightToLeftLayout_SetWithHandle_GetReturnsExpected(RightT
             Assert.Equal(expectedCreatedCallCount2, createdCallCount);
         }
 
+        [Fact]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TabControlTests.cs]:
```suggestion
        [WinFormsFact]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TabControlTests.cs [hunk 13]
```diff
@@ -1559,6 +1565,44 @@ public void TabControl_RightToLeftLayout_SetWithHandle_GetReturnsExpected(RightT
...
+        public void TabControl_RightToLeftLayout_SetWithHandler_CallsRightToLeftLayoutChanged()
+        {
+            var control = new TabControl
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TabControlTests.cs]:
```suggestion
            using var control = new TabControl
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemTests.cs [hunk 14]
```diff
@@ -350,24 +350,31 @@ public void ToolStripItem_AllowDrop_Set_GetReturnsExpected(bool value)
             // Set same.
             item.AllowDrop = value;
             Assert.Equal(value, item.AllowDrop);
+
+            // Set different.
+            item.AllowDrop = !value;
+            Assert.Equal(!value, item.AllowDrop);
         }
 
         [Theory]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemTests.cs]:
```suggestion
        [WinFormsTheory]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemTests.cs [hunk 15]
```diff
@@ -341,7 +341,7 @@ public void ToolStripItem_Alignment_SetInvalid_ThrowsInvalidEnumArgumentExceptio
         [CommonMemberData(nameof(CommonTestHelper.GetBoolTheoryData))]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripItemTests.cs]: WinFormsTheory

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanelTests.cs [hunk 16]
```diff
@@ -42,17 +42,24 @@ public void Ctor_Default()
 
         [Theory]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripPanelTests.cs]:
```suggestion
        [WinFormsTheory]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/VScrollBarTests.cs [hunk 17]
```diff
@@ -14,86 +14,92 @@ public class VScrollBarTests
         [Fact]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/VScrollBarTests.cs]:
```suggestion
        [WinFormsFact]
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/VScrollBarTests.cs [hunk 18]
```diff
@@ -14,86 +14,92 @@ public class VScrollBarTests
...
-            Assert.Equal(0, scrollBar.TabIndex);
-            Assert.False(scrollBar.TabStop);
-            Assert.Empty(scrollBar.Text);
-            Assert.Equal(0, scrollBar.Top);
-            Assert.Equal(0, scrollBar.Value);
-            Assert.True(scrollBar.Visible);
-            Assert.True(scrollBar.Width > 0);
-
-            Assert.False(scrollBar.IsHandleCreated);
+            var control = new SubVScrollBar();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/VScrollBarTests.cs]:
```suggestion
            using var control = new SubVScrollBar();
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs (hunk 2 above)
Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ControlTests.cs]: Not appropriate here as the test is `Control_AllowDrop_SetWithHandleNonSTAThread_ThrowsInvalidOperationException`

**Comment by RussKie** [comment]
🚀

**Pull request merged**